docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+
+- Review and standardize the usage of the concept of readiness and availability to align to K8s API conventions / 
+  conditions used in core K8s objects like `Pod`, `Node`, `Deployment`, `ReplicaSet` etc. 
+- Transition to K8s API conventions fully aligned conditions types/condition management (and thus deprecation of
+  the Cluster API "custom" guidelines for conditions).
+
+The last set of changes is a consequence of the above changes, or small improvements to address feedback received
+over time; changes in this group will be detailed case by case in the following paragraphs, a few examples:
+
+- Change the semantic of ReadyReplica counters to use Machine's Ready condition instead of Node's Ready condition.
+  (so everywhere Ready is used for a Machine it always means the same thing)
+- Add a new condition monitoring the status of the connectivity to workload clusters (`RemoteConnectionProbe`).
+
+In order to keep making progress on this proposal, the fist iteration will be focused on
+
+- Machines
+- MachineSets
+- MachineDeployments
+- MachinePools
+- KubeadmControlPlane (ControlPlanes)
+- Clusters
+
+Other resources will be added as soon as there will be agreement on the general direction.
+
+Overall, the union of all those changes, is expected to greatly improve status fields, conditions, replica counters 
+and print columns.
+
+Those improvements are expected to provide benefit to users interacting with the systems, using monitoring tools, and 
+building higher level systems or products on top of Cluster API.
+
+### Readiness and Availability
+
+The [condition CAEP](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20200506-conditions.md) in Cluster API introduced very strict requirements about “Ready” condition, mandating it 
+to exists on all resources and also mandating that Ready must be computed as the summary of all the to other existing
+conditions.
+
+However, over time Cluster API maintainers recognized several limitations of the “one fit all”, strict approach.
+
+e.g. Higher level abstractions in Cluster API are designed to remain operational during lifecycle operations,
+for instance a Machine deployment is operational even if is rolling out.
+
+But the use cases above where hard to combine with the strict requirement to have all the conditions true, and
+as a result today Cluster APi resources barely have conditions surfacing that lifecycle operations are happening, or where
+those condition are defined they have a semantic which is not easy to understand, like e.g. 'Resized' or 'MachinesSpecUpToDate'.
+
+e.g. when you look at higher level abstractions in Cluster API like Clusters, MachineDeployments and ControlPlanes, readiness
+might be confusing, because those resources usually accept a certain degree of not readiness, e.g. MachineDeployments are
+usually ok even if a few machine is not ready (up to MaxUnavailable).
```

Comment:
This is a great point, and thank you for calling it out. More often than not when we look for "Ready" we're looking for things to be healthy. Which might be better suited for a different signal.

Reply:
Thanks for the feedback, resolving to keep the number of open threads under control